tests/middleware/test_middleware.py

```diff
@@ -12,11 +12,25 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         await self.app(scope, receive, send)
 
 
-def test_middleware_repr() -> None:
-    middleware = Middleware(CustomMiddleware, "foo", bar=123)
-    assert repr(middleware) == "Middleware(CustomMiddleware, 'foo', bar=123)"
+def test_class_matches_middleware_protocol() -> None:
+    assert isinstance(CustomMiddleware, _MiddlewareFactory)
 
 
-def test_middleware_iter() -> None:
-    cls, args, kwargs = Middleware(CustomMiddleware, "foo", bar=123)
-    assert (cls, args, kwargs) == (CustomMiddleware, ("foo",), {"bar": 123})
+class TestMiddleware:
```

Comment:
Why this refactor? We have an existing test layout, modules with functions in them. I'm not going to argue what test layout is best, there's always tradeoffs, but changing the existing test layout of an established project like this is going to require a strong reason to do so.

Reply:
The first problem with the initial tests is creation of a new middleware object in each test. This is not needed. Doing this in thousands of tests results in performance issues (and as a result,  increased enviromental damage)

The second problem: difficulty with adding new tests, and as a result loss of control of the code and loss of value of the tests (I'm referring here `tests/middleware/test_base.py`)

Reply:
> as a result, increased enviromental damage
I think you are exaggerating here. Starlette's test suite is relatively small and quick to run, especially compared to other things in our industry nowadays like asking an AI to do things for you. And you could always just assign it to a global and call it a day, no need to introduce a new class

Reply:
@adriangb: Class was introduced for grouping and readability.

Reply:
all in all I'm OK with closing this PR, but lets see a bigger picture. I will be messing around with Starlette for a while :smile: And I will picking up a lot of stuff along the way, it is unlikely that I will not be introducing significant improvements to the code base, but small improvements is a good thing as well. :blush: